tests/test_python_support.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+"""
+Verifying that all the tools can be run even by older python versions.
+
+Uses `mypy --python-version 3.X <path>` output to check for substrings that
+indicate the type-hints in the code are not compatible with this version.
+"""
+
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+HERE = Path(__file__).resolve().parent
+ROOT_DIR = HERE.parent
+
+EXIT_CODE = 0
+
+os.chdir(ROOT_DIR)
+
+versions_to_check = [
+    "3.7",
+    "3.8",
+    "3.9",
+]
+
+dirs_to_check = [
+    "tools",
+    "common",
+    "core/tools",
+]
+
+signs_of_issues = [
+    "Module 'typing' has no attribute",  # we need to import some stuff from typing_extensions
```

Comment:
can't we get the same info from pyright, if the python version is set appropriately?

Reply:
It was indeed possible, changed in [29a778a](https://github.com/trezor/trezor-firmware/pull/2308/commits/29a778aca0103080a606ca39df559569ef408f10). It even correctly identified one more thing mentioned above. The only drawback is that it is slower than `mypy`, taking 30 seconds on my machine